src/app/components/category-products/category-products.component.html

```diff
@@ -0,0 +1,222 @@
+        <h5 class="mb-3">Categories</h5>
+        <ul class="nav nav-category" id="categoryCollapseMenu">
+          <li *ngFor="let category of categories$ | async" class="nav-item border-bottom w-100 collapsed">
+            <a [routerLink]="['/categories',category.id]" class="nav-link">{{category.name}}
+              <i class="feather-icon icon-chevron-right"></i>
+            </a>
+          </li>
+        </ul>
+      </aside>
+      <div class="col-12 col-lg-9 col-md-8 px-10">
+        <ng-container *ngIf="title$ | async as item"><!-- card -->
+          <div class="card mb-4 bg-light border-0">        <!-- card body -->
+            <div class="card-body p-9">
+              <h2 class="mb-0 fs-1">{{item.name}}</h2>
+            </div>
+          </div>
+        </ng-container>
+        <ng-container *ngIf="productsByCategoryId$ | async as products">
+          <div class="mb-3">
+            <p class="mb-0"><span class="text-dark">{{products.length}} </span> Products found </p>
+          </div>
+
+
+          <div class="card card-product mb-5" *ngFor="let product of products">
+            <!-- card body -->
+            <div class="card-body">
+              <div class=" row align-items-center">
+                <!-- col -->
+                <div class="col-md-4 col-12">
+
+                  <div class="text-center position-relative ">
+                    <a href="#!">
+                      <!-- img --><img src="{{product.imageUrl}}" alt="{{product.name}}" class="mb-3 img-fluid"></a>
+
+                  </div>
+                </div>
+                <div class="col-md-8 col-12 flex-grow-1">
+                  <!-- heading -->
+                  <ng-container *ngIf="title$ | async as category">
+                    <div class="text-small mb-1"><a href="#!"
+                                                    class="text-decoration-none text-muted"><small>{{category.name}}</small></a>
+                    </div>
+
+                  </ng-container>
+                  <h2 class="fs-6"><a href="#!"
+                                      class="text-inherit text-decoration-none">{{product.name}}</a>
+                  </h2>
+                  <div>
+                    <small class="text-warning"> <i *ngFor="let val of ratingValue$ | async" class="bi"
+                                                    [ngClass]="{'bi-star-fill': val === 1, 'bi-star': val === 0, 'bi-star-half': val === 0.5}"></i>
+                    </small><span class="text-muted small">{{product.ratingValue}} ({{product.ratingCount}})</span>
+
+                  </div>
+                  <div class=" mt-6">
+                    <!-- price -->
+                    <div><span class="text-dark">${{product.price}}</span>
+                    </div>
+
+                  </div>
+                </div>
+              </div>
+
+            </div>
+          </div>
+
+
+        </ng-container>
+      </div>
+    </div>
+  </div>
+</section>
+
+
+<footer class="footer mt-5">
```

Comment:
i think it is a duplicated code fragment - what is the better place to put it?

Reply:
I use other component